File: test-login.mjs
import { chromium } from 'playwright';

async function testLogin() {
  const browser = await chromium.launch({ headless: true });
  const context = await browser.newContext();
  const page = await context.newPage();

  // Listen for console errors
  page.on('console', msg => {
    if (msg.type() === 'error') {
      console.log('Console error:', msg.text());
    }
  });

  page.on('pageerror', err => {
    console.log('Page error:', err.message);
  });

  try {
    // Go to landing page
    console.log('1. Going to landing page...');
    await page.goto('http://localhost:3001', { waitUntil: 'networkidle' });
    console.log('   Landing page loaded');

    // Click Teacher button
    console.log('2. Clicking Teacher button...');
    await page.click('text=Teacher');
    await page.waitForURL('**/login**');
    console.log('   Navigated to login page');

    // Fill in login form
    console.log('3. Filling login form...');
    await page.fill('input[type="email"]', 'test@example.com');
    await page.fill('input[type="password"]', 'password123');

    // Submit
    console.log('4. Submitting login...');
    await page.click('button[type="submit"]');

    // Wait for navigation
    await page.waitForTimeout(2000);

    const url = page.url();
    console.log('   Current URL:', url);

    if (url.includes('/teacher')) {
      console.log('5. SUCCESS - Logged in and at teacher dashboard');
      // Check what errors appear on the dashboard
      const content = await page.textContent('body');
      if (content.includes('Loading')) {
        console.log('   Dashboard showing Loading... state');
      }
      if (content.includes('Unable to load')) {
        console.log('   Dashboard showing error state');
      }
    } else {
      console.log('5. FAILED - Not at teacher dashboard');
    }

  } catch (err) {
    console.log('Test error:', err.message);
  } finally {
    await browser.close();
  }
}

testLogin();

File: test_login.py
from playwright.sync_api import sync_playwright

def test_login():
    with sync_playwright() as p:
        browser = p.chromium.launch(headless=True)
        context = browser.new_context()
        page = context.new_page()

        # Listen for console errors
        def on_console(msg):
            if msg.type == 'error':
                print(f'Console error: {msg.text}')

        page.on('console', on_console)

        def on_page_error(err):
            print(f'Page error: {err}')

        page.on('pageerror', on_page_error)

        try:
            # Go to landing page
            print('1. Going to landing page...')
            page.goto('http://localhost:3001', wait_until='networkidle')
            url = page.url
            print(f'   Landing page loaded: {url}')

            # Click Teacher button (button containing text Teacher)
            print('2. Clicking Teacher button...')
            page.get_by_role('button', name='Teacher').click()
            page.wait_for_url('**/login**')
            url = page.url
            print(f'   Navigated to: {url}')

            # Fill in login form
            print('3. Filling login form...')
            page.fill('#email', 'test@example.com')
            page.fill('#password', 'password123')

            # Submit
            print('4. Submitting login...')
            page.click('button[type="submit"]')

            # Wait for navigation
            page.wait_for_timeout(3000)

            url = page.url
            print(f'   Current URL: {url}')

            if '/teacher' in url:
                print('5. SUCCESS - At teacher dashboard')
                content = page.text_content('body')
                if 'Loading' in content:
                    print('   Dashboard showing Loading state')
                if 'Unable to load' in content:
                    print('   Dashboard showing error state')
            else:
                print('5. NOT at teacher dashboard')
                content = page.text_content('body')[:500]
                print(f'   Page content preview: {content}')

        except Exception as err:
            print(f'Test error: {err}')
            import traceback
            traceback.print_exc()
        finally:
            browser.close()

if __name__ == '__main__':
    test_login()